### DISCUSSION TITLE
Using Splink without training data

### DISCUSSION THREAD

**Discussion by jcamatta** [moj-analytical-services/splink#2243] [Q&A] [answered]
Hi,

First, thank you very much for the awesome library and the detailed documentation. It has helped me a lot in understanding a difficult topic.

I am working on creating a master patient index that receives a JSON and checks the FHIR database for any existing patients who might be the same underlying real person, taking into account typo errors and other discrepancies. This is a service that needs to be fast. It will receive the JSON, perform an API search on the database (like a blocking phase), and then compare the received JSON with other patients in the database.

Initially, the database is empty, so my problem is that I don't have any data. Therefore, I can't train a Splink model. However, I don't want to have "two" codebases: one to use before the data is generated in production, and another to use with a trained Splink model. My question is whether I can use Splink with the comparison rules to check for duplicates even if I don't have training data to estimate the m and u parameters. Once I have the data, I will train a model and apply the same comparison rules with my estimated m and u parameters.

Initially, I will set the threshold very high, and as soon as I gather data and train a model, I will lower it.

Thank you very much for your help, and forgive my English; I did my best.

**Comment by RobinL** [comment]
To train a model, you really do need data.  Specifically a reasonably large (thousands of rows, ideally a million or more) dataset that is representative of the data you're going to apply the model to.

i.e. the usual approach to what you're trying to do is to train a model on some representative data, save the model out, and then load the pre-trained model into your live app and apply it to new data as it comes in.

**Reply by jcamatta** [reply]
Thank you for the fast reply.

Yes, that is what I was thinking. I will have a service whose task is to train a model and save it. Then, another one will load the model and make predictions between the incoming request and the database.

But, what happens if I make a prediction without training the model? Otherwise, I will need to create a rule-matching-based algorithm to predict whether two resources are duplicates until I gather enough data to train the model. Without data, I can't use the Fellegi and Sunter model, so I have no choice but to create some deterministic arbitrary approach.

**Reply by RobinL** [reply, marked as answer]
Splink does have built in default values, so `predict()` will work.  It just won't give you very good results until trained
